ui/src/app/components/confirm-workspace-delete-modal.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+
+import { ResourceType } from 'generated/fetch';
+
+import { findApp, UIAppType } from 'app/components/apps-panel/utils';
+import { Button } from 'app/components/buttons';
+import { TextInput } from 'app/components/inputs';
+import { WarningMessage } from 'app/components/messages';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
+import { appsApi } from 'app/services/swagger-fetch-clients';
+import { toDisplay } from 'app/utils/resources';
+
+export interface ConfirmWorkspaceDeleteModalProps {
+  closeFunction: Function;
+  receiveDelete: Function;
+  workspaceName: string;
+  workspaceNamespace: string;
+}
+export const ConfirmWorkspaceDeleteModal = ({
+  closeFunction,
+  receiveDelete,
+  workspaceName,
+  workspaceNamespace,
+}: ConfirmWorkspaceDeleteModalProps) => {
+  const [loading, setLoading] = useState<boolean>(false);
+  const [deleteDisabled, setDeleteDisabled] = useState<boolean>(true);
+  const [checkForApp, setCheckForApp] = useState<boolean>(true);
+  const [appsExistForWorkspace, setAppsExistForWorkspace] =
+    useState<boolean>(false);
+
+  useEffect(() => {
+    setCheckForApp(true);
+    appsApi()
+      .listAppsInWorkspace(workspaceNamespace)
+      .then((userApps) => {
+        setAppsExistForWorkspace(!!findApp(userApps, UIAppType.CROMWELL));
+        setCheckForApp(false);
+      })
+      .finally(() => {
+        setCheckForApp(false);
+      });
+  }, []);
+
+  const emitDelete = () => {
+    setLoading(true);
+    receiveDelete();
+  };
+
+  const validateDeleteText = (event) => {
+    setDeleteDisabled(!event.toLowerCase().match('delete'));
+  };
+
+  const displayUserInput = !checkForApp && !appsExistForWorkspace;
+
+  return (
+    <Modal loading={loading}>
+      <ModalTitle style={{ lineHeight: '28px' }}>
+        Warning — All work in this workspace will be lost.
+      </ModalTitle>
+      <ModalBody style={{ marginTop: '0.3rem', lineHeight: '28.px' }}>
+        <div>
+          <div>
+            Are you sure you want to delete {ResourceType.WORKSPACE} :{' '}
```

Comment:
This changes "Workspace" to "WORKSPACE" - so we should continue to use `toDisplay()` here

Reply:
or instead... we know it's a workspace so we can just say Workspace :)